src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -537,7 +552,12 @@ ReportAie::writeReport(const xrt_core::device* /*_pDevice*/,
       _output << std::endl;
       count = 0;
       for (auto& tile : graph.get_child("tile")) {
-        _output << boost::format("Core [%2d]\n") % count++;
+	int curr_core = count++;
+	if(aieCoreList.size() && (std::find(aieCoreList.begin(), aieCoreList.end(),

```

Comment:
Do you need the .size() test? std::find() should be fine with empty list?

Reply:
Actually if aieCoreList is empty then we should fail this if condition and print all the cores information. Hence the size check is required here. 
If not empty then we will try to find the specified core..

Reply:
If you are interested by emptiness, just use `aieCoreList.empty()`